### PR TITLE
Add preadd functionality

### PR DESCRIPTION
**Pull request by aadibajpai**
also fixed old react warnings about table elements not having unique keys

### CODE REVIEW

**Comment by aadibajpai** [comment]
pls qa 😩 

I think editing card might be nice but not sure how to implement and might be nice to have a status icon signifying if the user who was preadded had logged in or something

File: pages/api/preadd.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+	await dbConnect();
+
+	switch (req.method) {
+		case 'GET':
+			return res.send(await PreAdd.find());
+		case 'DELETE':
+			const { userId } = req.body;
+			console.log(`removing user with id ${userId} from preadded`);
+			const deleted = await PreAdd.findByIdAndDelete(userId);
```
[the remark this answers is not in the document]
Comment (aadibajpai) [reply pages/api/preadd.ts:19]: I think it's fine bc we check if the person is organizer or not

**Comment by aadibajpai** [comment]
also what is somewhat inconsistent I saw is that some components have their api handlers defined in the same file while some have them defined in OrganizerDash and passed down as props.

**Comment by nkalupahana** [comment]
I can't pre-add on the preview? Getting a 500 whenever I submit. Also, just at a glance, it would be good if we could make sure the cards are the same size.

**Comment by Ace314159** [comment]
I can add just fine, but it does fail when you add duplicate entries. And yeah, I also agree the cards should be the same size.

**Comment by nkalupahana** [comment]
Oh, you know what, I think I just tested with a duplicate instead of testing normally first. Yeah, that'd do it.

**Review by nkalupahana — APPROVED** [review]

**Comment by nkalupahana** [comment]
When you submit it says "Stored preadd information for undefined users" lol

**Comment by aadibajpai** [comment]
> When you submit it says "Stored preadd information for undefined users" lol

okay should be fine now, I changed it to use the returned object. Also added handling of an empty submit.

**Review by Ace314159 — APPROVED** [review]

**Pull request merged**